cfy_manager/networks/networks.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+#########
+# Copyright (c) 2018 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+
+import yaml
+import argh
+from os.path import join
+from cfy_manager.logger import get_logger
+from cfy_manager.utils import common
+from cfy_manager.utils.files import write_to_file
+from cfy_manager.constants import (
+    NETWORKS_DIR,
+    CERT_METADATA_FILE_PATH,
+    CLOUDIFY_USER,
+    CLOUDIFY_GROUP
+)
+
+from cfy_manager.utils.certificates import (
+    create_internal_certs,
+    load_cert_metadata
+)
+
+logger = get_logger('networks')
+SCRIPT_DIR = join(NETWORKS_DIR, 'scripts')
+REST_HOME_DIR = '/opt/manager'
+
+
+def _run_update_provider_context_script(args):
+    script_path = join(SCRIPT_DIR, 'add-networks-to-provider-context.py')
+
+    # Directly calling with this python bin, in order to make sure it's run
+    # in the correct venv
+    python_path = join(REST_HOME_DIR, 'env', 'bin', 'python')
+    cmd = [python_path, script_path, str(args)]
+
+    return common.sudo(cmd)
+
+
+def _update_metadata_file(networks):
+    """
+    Add the new networks to /etc/cloudify/ssl/certificate_metadata
+    :param networks: a dict containing the new networks
+    """
+    metadata = load_cert_metadata()
+    old_networks = metadata.get('networks', {})
+    new_networks = _merge_dicts(networks['networks'], old_networks)
```

Comment:
this actually takes only new networks from "networks".
if a network exist in both old and new it will remain with old value, which is good, but perhaps we should warn the user on his?

Reply:
oh, the script actually throws an error when trying to update
so the same validation should be in this script, then...